sound/soc/sof/sof-audio.c

```diff
@@ -270,14 +270,16 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 	struct snd_sof_widget *swidget = widget->dobj.private;
 	struct snd_soc_dapm_path *p;
 
-	if (!widget_ops[widget->id].ipc_unprepare || !swidget->prepared)
-		goto sink_unprepare;
+	/* no need to process sink */
+	if (!swidget->prepared)
+		return;
```

Comment:
> this is changing the behavior: before this patch in case of !swidget->prepared we did `unprepare all widgets in the sink paths`, now we don't ?

If I read it correctly, @RanderWang added the "no need to process sink" comment, but I don't see the corresponding code change.

Reply:
Right, the comment is there and there is a return to not process the sinks.
I guess it is OK?

Reply:
I think this is fine, if the swidget is prepared we clear the flat that it is no longer in prepared state (call ipc_unprepare if it is valid) and unprepare it's sink path.
If we hit an swidget which is unprepared, there is no need to unprepare it again.